test/builtin_dataset_mocks.py

```diff
@@ -61,27 +58,30 @@ def _parse_mock_info(self, mock_info):
 
         return mock_info
 
-    def prepare(self, home, config):
+    def prepare(self, home, **options):
         root = home / self.name
         root.mkdir(exist_ok=True)
 
-        mock_info = self._parse_mock_info(self.mock_data_fn(self.info, root, config))
+        mock_info = self._parse_mock_info(self.mock_data_fn(datasets.info(self.name), root, **options))
 
-        available_file_names = {path.name for path in root.glob("*")}
-        required_file_names = {resource.file_name for resource in self.dataset.resources(config)}
-        missing_file_names = required_file_names - available_file_names
-        if missing_file_names:
-            raise pytest.UsageError(
-                f"Dataset '{self.name}' requires the files {sequence_to_str(sorted(missing_file_names))} "
-                f"for {config}, but they were not created by the mock data function."
-            )
+        # # FIXME: We need to handle missing files here
```

Comment:
Before we could access the resources without creating the datapipe and thus without loading the resources. This is no longer possible in the new design. We need to think about a new way to check for missing files. Note that this only makes the testing logic more complex, but has no effect on the user. 

One solution would be to mock out `Dataset2.__init__()`. I'll see if that is possible.

Reply:
> access the resources without creating the datapipe and thus without loading the resources


If the resources are already mocked, is this actually a problem?

Reply:
You are right in that the test would also fail if we did nothing here. The issue is that the error message wouldn't be expressive so it is harder for a contributor to fix the problem. 

However, this is already resolved. The solution I mentioned above worked out great and so we need very little changes to the old checking logic.